### PR TITLE
Implementation for hemoglobin weighting and pregnancy observer stratification.

### PR DESCRIPTION
**Pull request by albrja**
### Description
- *Category*: Other, research change request
- *JIRA issue*: [MIC-3011](https://jira.ihme.washington.edu/browse/MIC-3011) 
- *Research reference*: https://github.com/ihmeuw/vivarium_research/pull/836

Updated data artifact to include WRA location weights and configured it as the default for hemoglobin weighting (PLW is still available).  Added maternal hemorrhage state stratification to the pregnancy observer.

### Verification and Testing
Ran psimulate successfully and make_results and observed correct stratification was there.

### CODE REVIEW

File: src/vivarium_gates_iv_iron/components/hemoglobin.py [hunk 1]
```diff
@@ -36,60 +35,32 @@ def setup(self, builder: Builder):
...
-        self.distribution_parameters = builder.value.register_value_producer(
-            "hemoglobin.exposure_parameters",
-            source=builder.lookup.build_table(
-                distribution_parameters,
-                key_columns=["sex", "country"],
-                parameter_columns=["age", "year"],
-            ),
-            requires_columns=["age", "sex", "country"],
-        )
+        self.   distribution_parameters = builder.value.register_value_producer("hemoglobin.exposure_parameters",
```
Comment (mattkappel) [inline src/vivarium_gates_iv_iron/components/hemoglobin.py]: This is a problem

File: src/vivarium_gates_iv_iron/components/observers.py [hunk 2]
```diff
@@ -806,7 +782,7 @@ def sub_components(self) -> List:
     @property
     def name(self):
         return "anemia_observer"
-
+
```
Comment (mattkappel) [inline src/vivarium_gates_iv_iron/components/observers.py]: Unnecessary white space.

File: src/vivarium_gates_iv_iron/data/loader.py [hunk 3]
```diff
@@ -66,6 +66,7 @@ def get_data(lookup_key: str, location: str) -> pd.DataFrame:
         data_keys.POPULATION.TMRLE: load_theoretical_minimum_risk_life_expectancy,
         data_keys.POPULATION.ACMR: load_standard_data,
         data_keys.POPULATION.PLW_LOCATION_WEIGHTS: get_pregnant_lactating_women_location_weights,
+        data_keys.POPULATION.WRA_LOCATION_WEIGHTS: get_wra_location_weights,
```
Comment (mattkappel) [inline src/vivarium_gates_iv_iron/data/loader.py]: Anticipating a nitpick: Should we change this to `get_women_reproductive_age_location_weights` or the above to `get_plw_location_weights`?

**Review by mattkappel — APPROVED** [review]

File: src/vivarium_gates_iv_iron/components/hemoglobin.py [hunk 4]
```diff
@@ -36,60 +35,35 @@ def setup(self, builder: Builder):
...
-        self.distribution_parameters = builder.value.register_value_producer(
-            "hemoglobin.exposure_parameters",
-            source=builder.lookup.build_table(
-                distribution_parameters,
-                key_columns=["sex", "country"],
-                parameter_columns=["age", "year"],
-            ),
-            requires_columns=["age", "sex", "country"],
-        )
+        self.distribution_parameters = builder.value.register_value_producer("hemoglobin.exposure_parameters",
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/components/hemoglobin.py]: Why did you change this formatting? It was so much more readable before.
Same for the blocks below.

File: src/vivarium_gates_iv_iron/components/hemoglobin.py [hunk 5]
```diff
@@ -204,3 +179,13 @@ def disability_weight(self, index: pd.Index) -> pd.Series:
             name="anemia_levels",
         )
         return anemia_levels.map(ANEMIA_DISABILITY_WEIGHTS)
+
+    def _get_location_weights(self, builder: Builder):
+        if metadata.USE_PLW_LOCATION_WEIGHTS:
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/components/hemoglobin.py]: put this in the config file, not in metadata.py

File: src/vivarium_gates_iv_iron/constants/metadata.py [hunk 6]
```diff
@@ -63,3 +63,5 @@ class __AgeGroup(NamedTuple):
 
 
 AGE_GROUP = __AgeGroup()
+
+USE_PLW_LOCATION_WEIGHTS = False
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/constants/metadata.py]: remove this and put it in the config file.

File: src/vivarium_gates_iv_iron/data/loader.py (hunk 3 above)
Comment (rmudambi) [reply src/vivarium_gates_iv_iron/data/loader.py]: I think I prefer the verbose method name, but yeah consistency would be nice.

**Review by rmudambi — CHANGES_REQUESTED** [review]

File: src/vivarium_gates_iv_iron/components/hemoglobin.py (hunk 4 above)
Comment (mattkappel) [reply src/vivarium_gates_iv_iron/components/hemoglobin.py]: This is an artifact of merge conflict resolution.

File: src/vivarium_gates_iv_iron/data/loader.py (hunk 3 above)
Comment (albrja) [reply src/vivarium_gates_iv_iron/data/loader.py]: Changing this.

File: src/vivarium_gates_iv_iron/components/hemoglobin.py (hunk 5 above)
Comment (albrja) [reply src/vivarium_gates_iv_iron/components/hemoglobin.py]: It has been added as a population configuration.
